### DISCUSSION TITLE
How to allow photo taking / screenshot on AR Android

### DISCUSSION THREAD

**Discussion by doom3214** [needle-tools/needle-engine-support#181] [Q&A]
If I am not mistaken, currently webXR is being used to load AR on Android instead of scene viewer. This doesn't allow me to take photo of the AR experience, it seems currently iOS is able to.

How do I implement this feature. I really need help on this. Thanks!

**Comment by hybridherbst** [comment]
Hi, you can add the WebARCameraBackground component to your WebXR object and then make a button that just takes a screenshot. The component adds the required permissions so that the user's scenery can be part of the picture.

**Reply by doom3214** [reply]
Sorry I forgotten to mention, I'm doing this in Unity and I don't know the typescript function for screenshot. I only know how to write it in native C#/Unity

**Reply by marwie** [reply]
Hi, to write code for Needle Engine you have to use Typescript or Javascript. We don't export any C# code from Unity.
You can find an introduction here: https://engine.needle.tools/docs/getting-started/typescript-essentials.html

**Reply by marwie** [reply]
@doom3214 for taking a screenshot we have a builtin function as well (including saving it to your device)

`import { screenshot } from "@needle-tools/engine"`

Let me know if that helps you
